pkg/maps/metricsmap/metricsmap_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package metricsmap
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+)
+
+type MetricsMapTestSuite struct{}
+
+var _ = Suite(&MetricsMapTestSuite{})
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+func (m *MetricsMapTestSuite) TestGetNumPossibleCPUsFromFile(c *C) {
+	tests := []struct {
+		input    string
+		expected int
+	}{
+		{"0", 1},
+		{"0-7", 8},
+		{"0,2-3", 3},
+		{"", 0},
+		{"foobar", 0},
+	}
+
+	for _, t := range tests {
+		f, err := ioutil.TempFile("", "")
+		c.Assert(err, IsNil)
+		defer func() { os.Remove(f.Name()) }()
```

Comment:
If `getNumPossibleCPUsFromFile()` received a reader of some sort, then you wouldn't need this logic.

I'm also not strictly sure whether `defer()`s will run if you perform `c.Assert()`. I feel like I've had problems with similar sorts of unit tests before where I ended up leaving files behind if the test fails.

Reply:
Just validated - if any `c.Assert` invoked after the `defer` fails, the `defer` will be run.

Anyway, I like the idea of passing `io.Reader` instead, so I will change accordingly.